projects/storefrontapp-e2e-cypress/cypress/support/utils/intercept.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * intercepts and alias a GET request
+ * @param alias intercept alias
+ * @param path  xhr relative path
+ * @param cmsPrefix if true, prefixes `path` with /occ/v2/{BASE_SITE}
+ * @returns alias
+ */
+export function interceptGet(
+  alias: string,
+  path: string,
+  cmsPrefix?: boolean
+): string {
+  return interceptAndAlias('GET', alias, path, cmsPrefix);
+}
+/**
+ * intercepts and alias a POST request
+ * @param alias intercept alias
+ * @param path  xhr relative path
+ * @param cmsPrefix if true, prefixes `path` with /occ/v2/{BASE_SITE}
+ * @returns alias
+ */
+export function interceptPost(
+  alias: string,
+  path: string,
+  cmsPrefix?: boolean
+): string {
+  return interceptAndAlias('POST', alias, path, cmsPrefix);
+}
+
+/**
+ * intercepts and alias a DELETE request
+ * @param alias intercept alias
+ * @param path  xhr relative path
+ * @param cmsPrefix if true, prefixes `path` with /occ/v2/{BASE_SITE}
+ * @returns alias
+ */
+export function interceptDelete(
+  alias: string,
+  path: string,
+  cmsPrefix?: boolean
+): string {
+  return interceptAndAlias('DELETE', alias, path, cmsPrefix);
+}
+
+/**
+ * Intercepts a request and creates an alias for it
+ * @param method  http method
+ * @param alias intercept alias
+ * @param path  xhr relative path
+ * @param cmsPrefix true if the call is for a CMS OCC API
```

Comment:
Two comments about the `cmsPrefix` in all the functions in this file:

- The param does not seem related to cms.  Perhaps we use a name that does not reference the cms? (like sitePrefix, baseSitePrefix or soomethinf else).
- Could a default value of `true` improve the design?  I presume the majority of calls need the site-prefix.

Reply:
Renamed to baseSitePrefix.
I marked it as `optional` so that you don't have to provide it every time. To add a default value of `true`, it can't be optional. And if I change the logic, then it will be more confusing I think. Thoughts?

Reply:
A parameter with a default value isn't optional anymore in the the function declaration, but when you call the function, you can omit the argument and the parameter will be assigned the default value, like this example:

```
function applyDiscount(price: number, discount: number = 0.05): number {
    return price * (1 - discount);
}

console.log(applyDiscount(100)); // 95
```

from https://www.typescripttutorial.net/typescript-tutorial/typescript-default-parameters/

Reply:
Got it. Addressed. Thanks for the explanation